lib/Transforms/ConvertToCiphertextSemantics/AssignLayout.cpp

```diff
@@ -396,12 +408,74 @@ Value implementUnpackOpForScalar(
   return splatOp.getResult();
 }
 
+std::string printRelation(const IntegerRelation &rel) {
+  std::string str;
+  llvm::raw_string_ostream os(str);
+  rel.print(os);
+  return str;
+}
+
+FailureOr<Value> implementAssignLayoutNew(
+    tensor_ext::AssignLayoutOp op, int64_t ciphertextSize,
+    ImplicitLocOpBuilder &builder,
+    const std::function<void(Operation *)> &createdOpCallback) {
+  NewLayoutAttr layout = dyn_cast<NewLayoutAttr>(op.getLayout());
+  if (!layout) {
+    return op.emitError()
+           << "Expected layout to be an IntegerRelation-style layout";
+  }
+
+  RankedTensorType dataSemanticType =
+      cast<RankedTensorType>(op.getValue().getType());
+  llvm::SmallVector<int64_t> dataSemanticShape;
+
+  IntegerRelation rel = layout.getIntegerRelation();
+  for (unsigned varPos = rel.getVarKindOffset(VarKind::Range);
+       varPos < rel.getNumRangeVars(); ++varPos) {
```

Comment:
I just noticed this bug: you are starting at the range offset and then the upper bound of this for loop is the numrangevars, so this loop will have size 0.